From fbffa310e578f6eda7241e30cc7ef69ee2c97d71 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 3 Jun 2022 10:16:32 +0200
Subject: [PATCH 5/5] Revert "block: fix bio_clone_blkg_association() to
 associate with proper blkcg_gq"

This reverts commit b040625a1b2fc57921de45991570c744bd00298e.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/blk-cgroup.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/block/blk-cgroup.c b/block/blk-cgroup.c
index 16703e3cc..8dfe62786 100644
--- a/block/blk-cgroup.c
+++ b/block/blk-cgroup.c
@@ -1906,8 +1906,12 @@ EXPORT_SYMBOL_GPL(bio_associate_blkg);
  */
 void bio_clone_blkg_association(struct bio *dst, struct bio *src)
 {
-	if (src->bi_blkg)
-		bio_associate_blkg_from_css(dst, bio_blkcg_css(src));
+	if (src->bi_blkg) {
+		if (dst->bi_blkg)
+			blkg_put(dst->bi_blkg);
+		blkg_get(src->bi_blkg);
+		dst->bi_blkg = src->bi_blkg;
+	}
 }
 EXPORT_SYMBOL_GPL(bio_clone_blkg_association);
 
-- 
2.36.1.203.g1bcf4f6271

